### ISSUE TITLE
animations don't lopp after importing .dae from blender in v3

### ISSUE THREAD

**Issue by razvanc-r** [godotengine/godot#10999]
**Operating system or device, Godot version, GPU Model and driver (if graphics related):**
Ubuntu 17.04 - Godot v3.0.alpha.custom_build.6d233c6

The animation loops correctly in the editor but now then the scene is played. Check out the attached project.

[godot-dbg.zip](https://github.com/godotengine/godot/files/1277958/godot-dbg.zip)

**Comment by kubecz3k** [comment]
related issue https://github.com/godotengine/godot/issues/6169

**Comment by tofuhead** [comment]
I noticed this too. I managed to start and loop it by hand when the game start, in the main scene:
[code]
get_node("bg/AnimationPlayer").get_animation("default").set_loop(true)
get_node("bg/AnimationPlayer").play("default")'
[/code]
this works. but, the animation loop is not perfect. it's nothing like I set up in Blender when for example cube rotates perfect 90deg, so when looped it looks like infinite rotation. when brought to Godot, the timing is all wrong, it turns sort of 70deg, then stops for awhile and jump to beginning. 

so I kind of dropped this and though "oh it's just not done yet at all" because all the problems, but maybe I now investigate a bit more and drop in a sample project. 

.b

**Comment by kubecz3k** [comment]
@tofuhead for me it sounds like your problems might be related to bake resolution of animations during export/import? Would be nice if you could attach your model

**Comment by hedgehog90** [comment]
This is still an issue.

**Comment by nagasawamaki** [comment]
me too...
using 3.0.5

**Comment by drewpotter** [comment]
I have experienced the same issue. I also tried the fix mentioned in the YouTube video to add the suffix of "-loop" but it still does not work for me. I would be happy to try and fix this myself if I get a chance to, as I am planning to use Godot at work for making games and also for godot-slides to make interactive gamified slides.

**Comment by glaforte** [comment]
I reproduced the issue, this morning, with the provided data.
I dug into the code and confirmed that suffixing your animation's name by "loop" fixes the issue.
Another way to fix the issue is to convert the DAE into a TSCN and then manually changing the loop flag.
This feels more like a documentation issue to me: the "loop" suffix 'feature' is not well known.
I'll see about proposing a documentation change for this page:
https://docs.godotengine.org/en/3.0/getting_started/workflow/assets/importing_scenes.html

**Comment by glaforte** [comment]
@drewpotter Do you have some specific data that you can upload, in a new issue, that I can look at? Right now, with the original reporter's data: I can easily fix the issue with the "loop" suffix. So, I'll be happy to look at your data if that suffix does not work with it.

**Comment by qichunren** [comment]
Confirm this bug still exist in 3.1 beta2 .

**Comment by drewpotter** [comment]
Sorry for my late response @glaforte @qichunren - I will test this with the latest version of Godot and I will create a new issue if I find any problems.

**Comment by AdamRatai** [comment]
**My solution -  I just made a copy of every animation file in Godot. I cannot delete originals (they went back after reopening file), but copies can be looped without a problem.**

I'm on Godot  3.1.1 (stable) on windows 10. 

I managed to move Mixamo animations to Godot, but when I was sure I have quite good pipeline, my final animations (dae exported from blender via better collada and converted to tscn) doesn't loop.
I'm still new in Godot, but I made it very carefully, using this tutorial:

https://godotengine.org/article/importing-3d-assets-blender-gamedevtv

When I made preliminary tests (one object with one animation) everything was OK, but final collada with few scenes had described problems.
It looks like Godot is not saving some settings in file (afer closing and reopening, animations seems to lost their loop flag)
I think problem can be related to cleaning mixamo armature naming using python script (https://github.com/godotengine/godot-blender-exporter/issues/43), or some bad tick when importing/exporting (I mean some animations enabling this problem, unseen for most of users). 

Maybe it is not a bug, but somewhat too complicated and not really explained workflow.
But As I understand I cannot save changes in original Collada file, but TSCN copy should be editable, and it looks like it sometimes doesn't.

**Comment by RoboClonk** [comment]
> It's stated in this video https://www.youtube.com/watch?v=9ZYyJlgSvhM that you have to sufix "-loop" the animation before exporting to make it loop in godot whit no code. Have not tested it miself yet.

This worked for me. You don't have to duplicate every animation.

**Comment by AdamRatai** [comment]
I'm sure you right. That was more than year ago. Godot is changing very fast, and Blender even faster, so everything I wrote above is obsolete. I started to use GLTF2 exporter and GGT blender addon, and I'm not using dae at all.

**Comment by RoboClonk** [comment]
Yes, but I did it the same way before. Just copying and setting the loop flag. I had to do this every time I changed my animations, which was a bit annoying and time consuming. I use GLTF2 too and the only way to make the loop flag "procedurally" was by renaming the blender action and NLA Stack to "..-loop".

**Comment by akien-mga** [comment]
Is this still reproducible in the current `master` branch and 3.5?

Also related: #34394.
